### PR TITLE
spi_loopback test on stm32F7 with NOCACHE memory

### PR DESCRIPTION
**Pull request by FRASTM**
This patch is running the tests/drivers/spi/spi_loopback on th stm32f7 target boards when the Cache memory control is disabled.
The test case(spi.c)  is modified to allocate SPI Tx and RX buffers in NOCACHE memory when the DMA is enabled on the stm32f7 soc series.

On nucleo_f746zg and nucleo_f767zi : SPI 1 is used for testing : DMA 2 channel 5 & 2 and request 3 is selected.
--> SPI 1  MISO and MOSI pins must be connected on the board to pass the test.

The **tests/drivers/dma/loop_transfer/src /test_dma_loop.c** gives the sequence to initialize the  nocached buffer  (when CONFIG_NOCACHE_MEMORY is set)

Signed-off-by: Francois Ramu francois.ramu@st.com

### CODE REVIEW

**Comment by FRASTM** [comment]
```
*** Booting Zephyr OS build zephyr-v2.6.0-122-g0d9d8a94f058  ***                
Running test suite test_spi                                                     
===================================================================             
START - test_spi_loopback                                                       
I: SPI test on buffers TX/RX 0x20020000/0x20020060                              
I: SPI test slow config                                                         
I: Start complete multiple                                                      
I: CS control inhibited (no GPIO device)                                        
I: Passed                                                                       
I: Start complete loop                                                          
I: Passed                                                                       
I: Start null tx                                                                
I: Passed                                                                       
I: Start half start                                                             
I: Passed                                                                       
I: Skip half end                                                                
I: Skip every 4                                                                 
I: SPI test fast config                                                         
I: Start complete multiple                                                      
I: CS control inhibited (no GPIO device)                                        
I: Passed                                                                       
I: Start complete loop                                                          
I: Passed                                                                       
I: Start null tx                                                                
I: Passed                                                                       
I: Start half start                                                             
I: Passed                                                                       
I: Skip half end                                                                
I: Skip every 4                                                                 
I: Start complete loop                                                          
I: CS control inhibited (no GPIO device)                                        
I: Passed                                                                       
I: Start complete loop                                                          
I: CS control inhibited (no GPIO device)                                        
I: Passed                                                                       
I: All tx/rx passed                                                             
 PASS - test_spi_loopback in 0.65 seconds                                       
===================================================================             
Test suite test_spi succeeded                                                   
===================================================================             
PROJECT EXECUTION SUCCESSFUL     
```

**Comment by FRASTM** [comment]
place all the sections to ` __section__(".nocache")) ` removing the `__section__(".nocache.dma"))`

**Review by tbursztyka — APPROVED** [review]

**Comment by FRASTM** [comment]
Rebase
This PR is setting Tx and Rx buffers in the NOCACHE memory area when SPI is using the DMA
This is done by the CONFIG_NOCACHE_MEMORY flag

**Comment by FRASTM** [comment]
rebase
do not change the buffer_tx/buffer_rx & buffer2_tx/buffer2_rx when the CONFIG_NOCACHE_MEMORY is not set

**Comment by tbursztyka** [comment]
anybody else to review this PR? @erwango @MaureenHelm

File: tests/drivers/spi/spi_loopback/src/spi.c [hunk 1]
```diff
@@ -38,12 +38,27 @@ struct spi_cs_control spi_cs = {
...
 #define BUF_SIZE 17
+#define BUF2_SIZE 36
+
+#if CONFIG_NOCACHE_MEMORY
+static const char TX_DATA[] = "0123456789abcdef\0";
+static __aligned(32) char buffer_tx[BUF_SIZE] __used
+	__attribute__((__section__(".nocache")));
+static __aligned(32) char buffer_rx[BUF_SIZE] __used
+	__attribute__((__section__(".nocache")));
+static const char TX2_DATA[] = "Thequickbrownfoxjumpsoverthelazydog\0";
```
Comment (MaureenHelm) [inline tests/drivers/spi/spi_loopback/src/spi.c:49]: Please convert `TX_DATA` and `TX2_DATA` to `tx_data` and `tx2_data` respectively

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
CI results are pretty old, please rebase

File: tests/drivers/spi/spi_loopback/src/spi.c (hunk 1 above)
Comment (FRASTM) [reply tests/drivers/spi/spi_loopback/src/spi.c]: done

**Comment by FRASTM** [comment]
rebase on f0af9275edbfacd6fce5a042a9ca758e4050bf10
change name TX_DATA, TX2_DATA  (upper to lower case)

**Review by erwango — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**